src/main/scala/org/broadinstitute/dsde/workbench/sam/service/ResourceService.scala

```diff
@@ -392,4 +392,14 @@ class ResourceService(private val resourceTypes: Map[ResourceTypeName, ResourceT
     _ <- IO.fromFuture(IO(fireGroupUpdateNotification(policyId)))
     } yield ()
   }
+
+  def listAllFlattenedResourceUsers(resourceId: FullyQualifiedResourceId): Future[Set[UserIdInfo]] = {
```

Comment:
do you wanna try change this signature into IO.
Grab me if you need some help :)